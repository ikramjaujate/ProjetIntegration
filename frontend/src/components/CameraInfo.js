import '../css/Grades.css';


const CameraInfo = ({name, allowed, notification}) => {
    return (
        <div className="row align-items-center p-1 m-2 bg-light rounded col-10 col-lg-7 test1">
            <div className="name-camera-grade col-9 test2">{name}</div>
            <div className="bg-notification rounded col-3 col-lg-2 test3" style={{backgroundColor:allowed ? "var(--camera-allow)" : "var(--camera-refuse)" }}>  
                <i className="bi bi-bell icon-notification test4" style={{color:notification ? "white" : allowed ? "var(--camera-allow)" : "var(--camera-refuse)"}}></i>
            </div>
        </div>
    );
}

export default CameraInfo;