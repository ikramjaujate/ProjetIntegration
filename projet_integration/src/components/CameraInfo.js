import PropTypes from 'prop-types';
import '../css/Grades.css';


// {/* <i className="bi bi-person-circle" style={{color:'red', fontSize:'2em'}}></i> */}
const CameraInfo = ({name, allowed, notification}) => {
    return (
        <div class="row p-1 m-2 bg-light rounded">
            <div class="col-md-4 align-self-center">{name}</div>
            <div class="col-md-2 offset-md-5 rounded" style={{backgroundColor:allowed ? "#90E093" : "#FF6060" }}>  
                {/* <i class="bi bi-bell" style={{color:"white", fontSize:'100%'}}></i> */}
                <i class="bi bi-bell" style={{color:notification ? "white" : allowed ? "#90E093" : "#FF6060" , fontSize:'100%'}}></i>
            </div>
        </div>
    );
}

// CameraInfo.propTypes = {
//     className : PropTypes.string,
//     id_div : PropTypes.string,
//     id_elem : PropTypes.string,
//     name : PropTypes.string,
//     value : PropTypes.string,
//     onClick : PropTypes.func,
// }

export default CameraInfo;