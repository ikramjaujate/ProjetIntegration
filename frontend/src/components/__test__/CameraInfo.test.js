import React from 'react'
import Enzyme, { shallow } from 'enzyme'
import Adapter from 'enzyme-adapter-react-16' ;
import 'jest-enzyme' 
import 'jest-styled-components'
import CameraInfo from '../CameraInfo.js' ;
Enzyme.configure({ adapter: new Adapter() })


describe('CameraInfo', () => {
    it('Snapshot - renders correctement', () => {
        const snapshot = renderer.create(<CameraInfo name="Lounge" allowed={false} notification={true}/>).toJSON();
        expect(snapshot).toMatchSnapshot();
    })

    it('Vérifier le nom de la camera passé en paramètre.', () => {
      const wrapper = shallow(<CameraInfo name="Lounge" allowed="" notification=""/>)
      const paragraph = wrapper.find('div.test2')
      expect(paragraph).toHaveLength(1)
      expect(paragraph.text()).toEqual('Lounge')
    })

    it('Vérifier que le booleen passé en param dans allowed renvoie le bon backgroud (1er condition).', () => {
      const wrapper = shallow(<CameraInfo name="" allowed="" notification=""/>)
      const paragraph = wrapper.find('div.test3')
      expect(paragraph).toHaveLength(1)
      expect(paragraph.prop('style')).toHaveProperty("backgroundColor","var(--camera-refuse)")
    })

    it('Vérifier que le booleen passé en param dans allowed renvoie le bon backgroud (2eme condition).', () => {
        const wrapper = shallow(<CameraInfo name="" allowed="true" notification=""/>)
        const paragraph = wrapper.find('div.test3')
        expect(paragraph).toHaveLength(1)
        expect(paragraph.prop('style')).toHaveProperty("backgroundColor","var(--camera-allow)")
    })

    it('Vérifier que le booleen passé en param dans notification renvoie la bonne couleur (1ere condition).', () => {
        const wrapper = shallow(<CameraInfo name="" allowed="" notification="true"/>)
        const paragraph = wrapper.find('i.test4')
        expect(paragraph).toHaveLength(1)
        expect(paragraph.prop('style')).toHaveProperty("color","white")
    })

    it('Vérifier que la couleur passé en param dans notification renvoie la bonne couleur (2eme condition).', () => {
        const wrapper = shallow(<CameraInfo name="" allowed="true" notification=""/>)
        const paragraph = wrapper.find('i.test4')
        expect(paragraph).toHaveLength(1)
        expect(paragraph.prop('style')).toHaveProperty("color","var(--camera-allow)")
    })

    it('Vérifier que la couleur passé en param dans notification renvoie la bonne couleur (3eme condition).', () => {
        const wrapper = shallow(<CameraInfo name="" allowed="" notification=""/>)
        const paragraph = wrapper.find('i.test4')
        expect(paragraph).toHaveLength(1)
        expect(paragraph.prop('style')).toHaveProperty("color","var(--camera-refuse)")
    })  
})
