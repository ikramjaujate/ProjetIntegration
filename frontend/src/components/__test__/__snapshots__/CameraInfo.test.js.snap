// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CameraInfo Snapshot - renders correctement 1`] = `
<div
  className="row align-items-center p-1 m-2 bg-light rounded col-10 col-lg-7"
>
  <div
    className="name-camera-grade col-9"
  >
    Lounge
  </div>
  <div
    className="bg-notification rounded col-3 col-lg-2"
    style={
      Object {
        "backgroundColor": "var(--camera-refuse)",
      }
    }
  >
    <i
      className="bi bi-bell icon-notification"
      style={
        Object {
          "color": "white",
        }
      }
    />
  </div>
</div>
`;
